Mathlib/Topology/StoneCech.lean

```diff
@@ -331,4 +332,44 @@ instance StoneCech.compactSpace : CompactSpace (StoneCech α) :=
   Quotient.compactSpace
 #align stone_cech.compact_space StoneCech.compactSpace
 
+theorem disjoint_closure_of_disjoint_closed [TopologicalSpace α] [NormalSpace α]
+    {s t : Set α} (hs : IsClosed s) (ht : IsClosed t) (hd: Disjoint s t) :
+    Disjoint (closure (stoneCechUnit '' s)) (closure (stoneCechUnit '' t)) := by
+  let ⟨⟨f, cf⟩, hfs, hft, hf⟩ := exists_continuous_zero_one_of_closed hs ht hd
+  let Z := ULift.{u} <| Set.Icc (0 : ℝ) 1
+  have : CompactSpace Z := Homeomorph.ulift.symm.compactSpace
+  haveI : T2Space Z := Homeomorph.ulift.symm.t2Space
+  let g : α → Z := fun y' => ⟨f y', hf y'⟩
+  have hg : Continuous g := continuous_uLift_up.comp (cf.subtype_mk hf)
+  let uu := stoneCechExtend hg
+  have subs : ∀ (x : ℝ) (h : x ∈ Set.Icc (0 : ℝ) 1),
+      closure (stoneCechUnit '' (g ⁻¹' {⟨x, h⟩})) ⊆ uu ⁻¹' {⟨x, h⟩} := by
+    intros x hx
+    have closed_image : closure (uu ⁻¹' {⟨x, hx⟩}) = uu ⁻¹' {⟨x, hx⟩} := by
+      rw [closure_eq_iff_isClosed]
+      apply_rules [IsClosed.preimage, continuous_stoneCechExtend,
+        IsCompact.isClosed, isCompact_singleton, continuous_def.2]
+      simp only [preimage_id', imp_self, forall_const]
+    rw [←closed_image]
+    apply closure_mono
+    rw [←stoneCechExtend_extends hg, preimage_comp, image_subset_iff]
+  have closureSub: ∀ (x : ℝ), ∀ (u : Set α), (h0: (∀ (o : α), (o ∈ u)  → f o = x))
+      → (h : (x ∈ Set.Icc (0 : ℝ) 1)) → closure (stoneCechUnit '' u) ⊆ uu ⁻¹' {⟨x, h⟩} := by
+    intros x u hu xicc _
+    apply Subset.trans _ (subs x xicc)
```

Comment:
https://github.com/leanprover-community/mathlib4/pull/7750#discussion_r1364483179

Reply:
I'm afraid I don't really understand these comments involving `show_term` @alexjbest, could you elaborate a bit on how that tactic is used?

Reply:
Nice, that's a really nice tip!